src/components/DomainSelector.vue

```diff
@@ -16,22 +16,31 @@
   </div>
 </template>
 
-<script>
-export default {
-  name: 'domain-selector',
-  props: {
-    domainsList: Array,
-    selectedDomain: String,
-  },
-  methods: {
-    isDomainSelected(name) {
-      return this.selectedDomain === name;
-    },
-    select(newDomain) {
-      this.$emit('selectedDomain', newDomain);
-    },
-  },
-};
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Prop } from 'vue-property-decorator'
+
+@Component({
+  name: 'domain-selector'
+})
+export default class DomainSelector extends Vue {
+  @Prop({
+    default: () => [],
```

Comment:
Why not `deafult: []` directly? Is it a matter of mutability and default being evaluated once and for all?

Reply:
Yes, otherwise it's shared among every instance of this component: https://vuejs.org/v2/api/#props

Reply:
:+1: